base_partner_merge_user_consolidation/wizard/base_partner_merge.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2025 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class MergePartnerAutomatic(models.TransientModel):
+    _inherit = "base.partner.merge.automatic.wizard"
+
+    def _merge(self, partner_ids, dst_partner=None, extra_checks=True):
+        """Merge partners, assign a single user"""
+        super()._merge(partner_ids, dst_partner=dst_partner, extra_checks=extra_checks)
+        # Recompute dst_partner
+        Partner = self.env["res.partner"]
+        partner_ids_rs = Partner.browse(partner_ids).exists()
+        if not partner_ids_rs or len(partner_ids_rs) < 1:
```

Comment:
You can add a test for this situation by calling the method with only one partner, and then make sure nothing happens.